mediatek ramips: add support for Asus RT-N600

russinnes opened the pull request:
The ASUS RT-N600 Wireless Router is based on the MT7628AN SoC.

- CPU: MediaTek MT7628AN @580MHz WiSoC (2T2R 802.11bgn)
- WLAN: MediaTek MT7612EN (2T2R 802.11an -> 802.11an/ac)
- RAM: 64MB (DDR2), Flash 16MB (SPI, 16-pin)
- USB 2.0
- LED (ETHx4, WLANx2, USB, Power)
- PORTS: 4x 100M RJ45 LAN, 1x 100M WAN

Initial Flashing:

Download the "Asus Firmware Recovery Tool" from the Asus website
Set a static IP of 192.168.1.10 on a wired connection (Netmask 255.255.255.0)
Load the desired firmware file in the recovery tool (.bin, .trx, etc)
Power on the device while holding the reset button (Power LED will begin flashing when ready)
Update the firmware (recovery) in the software

Known Issues *: All LED's work and are configurable with the exception of the 5g WLAN LED.

Signed-off-by: Russ Innes [russ.innes@gmail.com](mailto:russ.innes@gmail.com)

russinnes committed:
ramips: add support for Asus RT-N600

The ASUS RT-N600 Wireless Router is based on the MT7628AN SoC.

CPU: MediaTek MT7628AN @580MHz WiSoC (2T2R 802.11bgn)
WLAN: MediaTek MT7612EN (2T2R 802.11an -> 802.11an/ac)
RAM: 64MB (DDR2), Flash 16MB (SPI, 16-pin)
USB 2.0
LED (ETHx4, WLANx2, USB, Power)
PORTS: 4x 100M RJ45 LAN, 1x 100M WAN

Initial Flashing:

Download the "Asus Firmware Recovery Tool" from the Asus website
Set a static IP of 192.168.1.10 on a wired connection (Netmask 255.255.255.0)
Load the desired firmware file in the recovery tool (.bin, .trx, etc)
Power on the device while holding the reset button (Power LED will begin flashing when ready)
Update the firmware (recovery) in the software

Known Issues *: All LED's work and are configurable with the exception of the 5g WLAN LED.
Signed-off-by: Russ Innes <russ.innes@gmail.com>

russinnes committed:
ramips: add support for Asus RT-N600

The ASUS RT-N600 Wireless Router is based on the MT7628AN SoC.

This is a config "clone" of the ASUS RT-AC1200v1, with a change to the model name definition basically for the end-user. This came via some trial and error with the AC1200 image discussed in #10083. The RT-N600 functions perfectly using the image from the AC1200, less the incorrect model definition.

CPU: MediaTek MT7628AN @580MHz WiSoC (2T2R 802.11bgn)
WLAN: MediaTek MT7612EN (2T2R 802.11an -> 802.11an/ac)
RAM: 64MB (DDR2), Flash 16MB (SPI, 16-pin)
USB 2.0
LED (ETHx4, WLANx2, USB, Power)
PORTS: 4x 100M RJ45 LAN, 1x 100M WAN
Initial Flashing:

Download the "Asus Firmware Recovery Tool" from the Asus website
Set a static IP of 192.168.1.10 on a wired connection (Netmask 255.255.255.0)
Load the desired firmware file in the recovery tool (.bin, .trx, etc)
Power on the device while holding the reset button (Power LED will begin flashing when ready)
Update the firmware (recovery) in the software

Signed-off-by: Russ Innes russ.innes@gmail.com

russinnes commented:
@therealsummoner here's the works in a single commit. Couldn't get it to squash on the branch I was working with, and screwed up my local git repo while doing so. So a new PR.

therealsummoner commented:
> @therealsummoner here's the works in a single commit. Couldn't get it to squash on the branch I was working with, and screwed up my local git repo while doing so. So a new PR.

A new PR can be problematic in most cases, because the history/previous changes are separated, so its better to avoid that later on.

csharper2005 commented:
Why the commit message is empty?

hauke commented on target/linux/ramips/dts/mt7628an_asus_rtn600.dts:
```diff
@@ -0,0 +1,134 @@
+		};
+
+
+
```
Please remove some of these empty lines

csharper2005 commented:
@russinnes hi! You probably need to squash 3 commits into the one.

csharper2005 commented:
@russinnes You can drop all after the SoB string.
```
Signed-off-by: Russ Innes <russ.innes@gmail.com>

**Update mt7628an_asus_rtn600.dts

Removed empty lines

Update mt7628an_asus_rtn600.dts

Remove empty lines**
```

w43322 commented on target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:
```diff
@@ -70,6 +70,10 @@ ramips_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0:lan" "1:lan" "2:lan" "3:lan" "4:wan" "6@eth0"
 		;;
+    asus,rtn600)
```
this can be merged with existing config (rt-ac1200 and etc.)

w43322 replied:
also you should probably use tabs instead of spaces

w43322 commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -84,6 +84,14 @@ define Device/asus_rt-n12-vp-b1
+  IMAGE_SIZE := 7872k
+  DEVICE_VENDOR := ASUS
+  DEVICE_MODEL := RTN600
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 
```
probably missing kmod-usb-ohci & kmod-usb-ledtrig-usbport

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rtn600.dts:
```diff
@@ -0,0 +1,127 @@
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "asus,rtn600", "mediatek,mt7628dan";
```
should be asus,rt-n600

w43322 commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -84,6 +84,14 @@ define Device/asus_rt-n12-vp-b1
 endef
 TARGET_DEVICES += asus_rt-n12-vp-b1
 
+define Device/asus_rtn600
```
ditto, missing dash

w43322 commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -84,6 +84,14 @@ define Device/asus_rt-n12-vp-b1
+define Device/asus_rtn600
+  IMAGE_SIZE := 7872k
+  DEVICE_VENDOR := ASUS
+  DEVICE_MODEL := RTN600
```
ditto

w43322 commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -84,6 +84,14 @@ define Device/asus_rt-n12-vp-b1
+  DEVICE_MODEL := RTN600
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 
+endef
+TARGET_DEVICES += asus_rtn600
```
ditto

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rtn600.dts:
```diff
@@ -0,0 +1,127 @@
+		led-upgrade = &led_sys;
+	};
+
+	leds {
```
possibly missing some leds? there're 9 leds in total according to user's manual:
https://dlcdnets.asus.com/pub/ASUS/wireless/RT-N600/E11273_RT_N600_Manual.pdf

w43322 commented on target/linux/ramips/dts/mt7628an_asus_rtn600.dts:
```diff
@@ -0,0 +1,127 @@
+
+&state_default {
+	gpio {
+		groups = "gpio", "i2c", "i2s", "refclk", "uart1", "wdt", "wled_an";
```
not all of those gpios are in use, please refer to the manual of the mt76x8 SoC and remove the unnecessary ones

w43322 reviewed (commented):
This device is very similar to RT-AC1200. Maybe you can check the dts files and commit message of that device for some insight.
I don't know if those devices (this with RT-AC1200 v1 and v2) should be merged into a single .dtsi file, but their commonalities seems plenty enough to justify a merge.

w43322 commented:
Second thought:

After looking at the internal photos of this device, I'm 90% sure it's the same internally as the RT-AC1200. Not 100% sure because I left it at campus.

So maybe a simple alt-name addition would suffice.

Can you try flashing the RT-AC1200 OpenWrt firmware on your device and see if everything works as intended?

russinnes commented:
> 

Good call, worked like a charm. I've tested everything obvious, everything working exactly as expected. 2xWLAN, switch config looks good, even the 5g LED. I flashed r20022-fd2f3136ca. I would agree at this point an alt-name would do the trick.

russinnes commented:
> > Good call, worked like a charm. I've tested everything obvious, everything working exactly as expected. 2xWLAN, switch config looks good, even the 5g LED. I flashed r20022-fd2f3136ca. I would agree at this point an alt-name would do the trick.
> 
> Glad to hear! Well, I guess Asus was too lazy to design new hardware for each product category. I wish I knew it earlier, so that I won't need to spend the premium on the RT-AC1200 :D

Is it worthwhile cloning the AC1200 for the N600 with a change in the DTS model definition to reflect as much? Might be nice for those users to see the appropriate name, but thats about the only difference.

w43322 commented:
> Is it worthwhile cloning the AC1200 for the N600 with a change in the DTS model definition to reflect as much? Might be nice for those users to see the appropriate name, but thats about the only difference.

Maybe it's worth it. Just create a new .dts file for the rt-n600 and include `rt-ac1200.dts`, then override the `compatibal` node by adding
```c
/ {
	compatible = "asus,rt-n600", "mediatek,mt7628dan";
	model = "ASUS RT-N600";
};
```to the file.

Also don't forget to make changes to `image.mk`.

russinnes committed:
ramips: add support for Asus RT-N600

The ASUS RT-N600 Wireless Router is based on the MT7628AN SoC.

This is a config "clone" of the ASUS RT-AC1200v1, with a change to the model name definition basically for the end-user. This came via some trial and error with the AC1200 image discussed in #10083. The RT-N600 functions perfectly using the image from the AC1200, less the incorrect model definition.

CPU: MediaTek MT7628AN @580MHz WiSoC (2T2R 802.11bgn)
WLAN: MediaTek MT7612EN (2T2R 802.11an -> 802.11an/ac)
RAM: 64MB (DDR2), Flash 16MB (SPI, 16-pin)
USB 2.0
LED (ETHx4, WLANx2, USB, Power)
PORTS: 4x 100M RJ45 LAN, 1x 100M WAN
Initial Flashing:

Download the "Asus Firmware Recovery Tool" from the Asus website
Set a static IP of 192.168.1.10 on a wired connection (Netmask 255.255.255.0)
Load the desired firmware file in the recovery tool (.bin, .trx, etc)
Power on the device while holding the reset button (Power LED will begin flashing when ready)
Update the firmware (recovery) in the software

Signed-off-by: Russ Innes russ.innes@gmail.com

russinnes committed:
ramips: add support for Asus RT-N600

This is a config "clone" of the ASUS RT-AC1200v1, with a change to the model name definition basically for the end-user. This came via some trial and error with the AC1200 image discussed in #10083. The RT-N600 functions perfectly using the image from the AC1200, less the incorrect model definition.

CPU: MediaTek MT7628AN @580MHz WiSoC (2T2R 802.11bgn)
WLAN: MediaTek MT7612EN (2T2R 802.11an -> 802.11an/ac)
RAM: 64MB (DDR2), Flash 16MB (SPI, 16-pin)
USB 2.0
LED (ETHx4, WLANx2, USB, Power)
PORTS: 4x 100M RJ45 LAN, 1x 100M WAN
Initial Flashing:

Download the "Asus Firmware Recovery Tool" from the Asus website
Set a static IP of 192.168.1.10 on a wired connection (Netmask 255.255.255.0)
Load the desired firmware file in the recovery tool (.bin, .trx, etc)
Power on the device while holding the reset button (Power LED will begin flashing when ready)
Update the firmware (recovery) in the software

Signed-off-by: Russ Innes russ.innes@gmail.com

russinnes committed:
Merge branch 'master' of https://github.com/russinnes/openwrt

russinnes committed:
Revert "Merge branch 'master' of https://github.com/russinnes/openwrt"

This reverts commit 32e0873c3d52df1d8dba92abb176e23b31211db8, reversing
changes made to e0af8cd4549da90e464639c392799a96ff4e64c5.

russinnes commented:
Closing and opening a new PR as the scope of this has changed since identifying the AC1200 image functionality.
